hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/DatanodeConfiguration.java

```diff
@@ -536,6 +541,24 @@ public void setWaitOnAllFollowers(boolean val) {
   )
   private int autoCompactionSmallSstFileNum = 512;
 
+  @Config(key = "rocksdb.auto-compaction-small-sst-file.interval.minutes",
+      defaultValue = "120",
+      type = ConfigType.LONG,
+      tags = { DATANODE },
+      description = "Auto compact small SST files interval in minutes."
+  )
+  private long autoCompactionSmallSstFileIntervalMinutes =
+      AUTO_COMPACTION_SMALL_SST_FILE_INTERVAL_MINUTES_DEFAULT;
+
+  @Config(key = "rocksdb.auto-compaction-small-sst-file.executors",
+      defaultValue = "1",
```

Comment:
Current RocksDB compaction is run in volume checker, which uses a unlimited thread pool( Executors.newCachedThreadPool). Not sure whether by default 1 thread will have performance impact to DN or not.

Reply:
Executors.newScheduledThreadPool is a  unlimited thread pool.
```
// java.util.concurrent.ScheduledThreadPoolExecutor#ScheduledThreadPoolExecutor(int, java.util.concurrent.ThreadFactory)
public ScheduledThreadPoolExecutor(int corePoolSize,
                                       ThreadFactory threadFactory) {
        super(corePoolSize, Integer.MAX_VALUE, 0, NANOSECONDS,
              new DelayedWorkQueue(), threadFactory);
}
```

Reply:
Hi @ChenSammi. Do you want to take a look?  Thanks.

Reply:
It should be ok to have one thread that visits each volume. Depends on how much time each compaction takes. Do we have any rough numbers for compaction duration?